clippy_lints/src/redundant_clone.rs

```diff
@@ -250,6 +275,32 @@ fn find_stmt_assigns_to<'a, 'tcx: 'a>(
     })
 }
 
+fn base_local<'tcx>(
```

Comment:
It would be nice to add a small comment that describes what this function does and why it's needed. If I understand it correctly, maybe `is_base_local_with_drop` is a better name?

Reply:
The function obtains moved-out-ablity of `place` during extraction steps of base `Local`, so there's no logical connection between two values returned... I'm not sure how should I name it.